src/test/java/com/google/api/codegen/testdata/csharp_gapic_resourcenames_library.baseline

```diff
@@ -0,0 +1,685 @@
+
+        /// <summary>
+        /// The bookId ID. Never <c>null</c>.
+        /// </summary>
+        public string BookIdId { get; }
+
+        /// <inheritdoc />
+        public override string ToString() => s_template.Expand(ArchivePathId, BookIdId);
+    }
+
+    /// <summary>
+    /// Resource name for a return.
+    /// </summary>
+    public sealed partial class ReturnName : ISingleResourceName
+    {
+        private static readonly PathTemplate s_template = new PathTemplate("shelves/{shelf}/books/{book}/returns/{return}");
+
+        /// <summary>
+        /// Parses a return resource name in string form into a <see cref="ReturnName"/>.
+        /// </summary>
+        /// <param name="returnName">The return resource name in string form. Must not be <c>null</c>.</param>
+        public static ReturnName Parse(string returnName)
+        {
+            GaxPreconditions.CheckNotNull(returnName, nameof(returnName));
+            ResourceName resourceName = s_template.ParseName(returnName);
+            return new ReturnName(resourceName[0], resourceName[1], resourceName[2]);
+        }
+
+        public static bool TryParse(string returnName, out ReturnName result)
+        {
+            GaxPreconditions.CheckNotNull(returnName, nameof(returnName));
+            ResourceName resourceName;
+            if (s_template.TryParseName(returnName, out resourceName))
+            {
+                result = new ReturnName(resourceName[0], resourceName[1], resourceName[2]);
+                return true;
+            }
+            else
+            {
+                result = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Constructs a return name from its component parts.
+        /// </summary>
+        /// <param name="shelfId">The shelf ID. Must not be <c>null</c>.</param>
+        /// <param name="bookId">The book ID. Must not be <c>null</c>.</param>
+        /// <param name="returnId">The return ID. Must not be <c>null</c>.</param>
+        public ReturnName(string shelfId, string bookId, string returnId)
+        {
+            ShelfId = GaxPreconditions.CheckNotNull(shelfId, nameof(shelfId));
+            BookId = GaxPreconditions.CheckNotNull(bookId, nameof(bookId));
+            ReturnId = GaxPreconditions.CheckNotNull(returnId, nameof(returnId));
+        }
+
+        /// <summary>
+        /// The shelf ID. Never <c>null</c>.
+        /// </summary>
+        public string ShelfId { get; }
+
+        /// <summary>
+        /// The book ID. Never <c>null</c>.
+        /// </summary>
+        public string BookId { get; }
+
+        /// <summary>
+        /// The return ID. Never <c>null</c>.
+        /// </summary>
+        public string ReturnId { get; }
+
+        /// <inheritdoc />
+        public override string ToString() => s_template.Expand(ShelfId, BookId, ReturnId);
+    }
+
+    public sealed partial class BookNameOneof : IResourceName
```

Comment:
Is the name of this controlled in gapic? I think we'll want to experiment with what feels natural here.